tests/CMakeLists.txt

```diff
@@ -0,0 +1,19 @@
+# Add test files
+set(TEST_FILES
+unique_pointer.t.cpp
+shared_pointer.t.cpp
+)
+# Create executable for tests
+add_executable(run_tests ${TEST_FILES})
+
+target_include_directories(SmartPointers PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/../SmartPointers)
+
+# Link Google Test to your test executable
+target_link_libraries(run_tests 
+gtest 
```

Comment:
It's ok in this case, but you don't need to necessarily have google-test. You could just use plain `assert` and put them in functions.

Reply:
done